packages/client/src/net/peer/peer.ts

```diff
@@ -108,28 +168,25 @@ export abstract class Peer extends EventEmitter {
 
     if (protocol.name === 'eth') {
       bound = new BoundEthProtocol(boundOpts)
+
+      await bound!.handshake(sender)
+
+      this.eth = <BoundEthProtocol>bound
     } else if (protocol.name === 'les') {
       bound = new BoundLesProtocol(boundOpts)
-    } else if (protocol.name === 'snap') {
-      bound = new BoundSnapProtocol(boundOpts)
-    } else {
-      throw new Error(`addProtocol: ${protocol.name} protocol not supported`)
-    }
 
-    // Handshake only when snap, else
-    if (protocol.name !== 'snap') {
       await bound!.handshake(sender)
-    } else {
-      if (sender.status === undefined) throw Error('Snap can only be bound on handshaked peer')
-    }
 
-    if (protocol.name === 'eth') {
-      this.eth = <BoundEthProtocol>bound
+      this.les = <BoundLesProtocol>bound
     } else if (protocol.name === 'snap') {
+      bound = new BoundSnapProtocol(boundOpts)
+      if (sender.status === undefined) throw Error('Snap can only be bound on handshaked peer')
+
       this.snap = <BoundSnapProtocol>bound
-    } else if (protocol.name === 'les') {
-      this.les = <BoundLesProtocol>bound
+    } else {
+      throw new Error(`addProtocol: ${protocol.name} protocol not supported`)
     }
```

Comment:
This looks wild but is only a small local refactor making things a bit cleaner. 🙂